pkg/terraform/state/handler.go

```diff
@@ -66,7 +66,57 @@ type handler struct {
 	jobs                batchv1.JobController
 }
 
+func (h *handler) OnChange(key string, obj *v1.State) (*v1.State, error) {
+	logrus.Debug("State On Change Handler")
+	if obj == nil {
+		return nil, nil
+	}
+
+	if obj.DeletionTimestamp != nil {
+		return nil, nil
+	}
+
+	if obj.Spec.Version < 1 {
+		obj.Spec.Version = 1
+	}
+	if obj.Spec.Image == "" {
+		obj.Spec.Image = fmt.Sprintf("%s:latest", DefaultExecutorImage)
+	}
+
+	input, ok, err := h.gatherInput(obj)
+
+	if !ok {
+		v1.ExecutionConditionMissingInfo.SetStatus(obj, err.Error())
+
+		return h.states.Update(obj)
+	}
+	if err != nil {
+		return obj, err
+	}
+
+	v1.ExecutionConditionMissingInfo.False(obj)
+	v1.ExecutionConditionJobDeployed.False(obj)
+
+	return tfv1.UpdateStateOnChange(func(state runtime.Object) (runtime.Object, error) {
```

Comment:
This... is not returning just the *v1.State. This needs to be broken up.

Reply:
refactored.